### PR TITLE
[Task][66491] Fetch and display data on Home Screen

### PR DESCRIPTION
**Pull request by Lomahs1401**
## Related Tickets
- [#Task 66491: Fetch and display data on Home Screen](https://edu-redmine.sun-asterisk.vn/issues/66491)
## WHAT
Fetch data from API and display to the UI
 - Image 
 - Name Recipe
 - Serving
 - Lịkes
 - Estimate time
## Evidence (Screenshot or Video)
image
image

https://github.com/awesome-academy/FoodRecipeApp/assets/77973385/da5ac21a-1258-4485-9c8c-5b61c0caab6e

## Review Checklist

Category | View Point | Description | Self review | Reviewer2 (name)
--- | --- | --- | --- | ---
Conventions | Sun* coding conventions | https://sal.vn/WtYPAy |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Conventions | Kotlin coding conventions | https://kotlinlang.org/docs/coding-conventions.html |<li>- [ ] yes</li>|<li>- [ ] yes</li>  
Redmine | Sun* Redmine working process  | https://sal.vn/piPoQn |<li>- [ ] yes</li>|<li>- [ ] yes</li>  

## Notes (Optional)
*(Impacted Areas in Application(List features, api, models or services that this PR will affect))*

*(List gem, library third party add new)*

*(Other notes)*

### CODE REVIEW

File: app/src/main/java/com/example/foodrecipeapp/data/repo/source/remote/fetchjson/GetJsonFromUrl.kt [hunk 1]
```diff
@@ -66,6 +88,8 @@ class GetJsonFromUrl<T : Any> constructor(
             responseData.append(it.trim())
         }
 
+        Log.e("CallAPI", "Query Success: $responseData")
+
```
Comment (ghost) [inline app/src/main/java/com/example/foodrecipeapp/data/repo/source/remote/fetchjson/GetJsonFromUrl.kt]: xóa log e

File: app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt [hunk 2]
```diff
@@ -33,6 +35,10 @@ class HomeFragment :
     override fun initView() {
         homeChildAdapter.setData(getListHomeChild())
         binding.rcvHomeParent.adapter = homeChildAdapter
+
+        dialog = ProgressDialog(context)
+        dialog.setTitle("Loading...")
```
Comment (ghost) [inline app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt]: text trên màn hình bỏ vô string.xml

File: app/src/main/res/layout/layout_search_home_child.xml [hunk 3]
```diff
@@ -46,7 +46,7 @@
     <com.google.android.material.button.MaterialButton
         android:layout_width="@dimen/dp_70"
         android:layout_height="wrap_content"
-        android:layout_marginEnd="@dimen/dp_28"
+        android:layout_marginEnd="16dp"
```
Comment (ghost) [inline app/src/main/res/layout/layout_search_home_child.xml]: dùng file dimen

Comment (Lomahs1401) [reply app/src/main/res/layout/layout_search_home_child.xml]: Done

File: app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt (hunk 2 above)
Comment (Lomahs1401) [reply app/src/main/java/com/example/foodrecipeapp/screen/home/HomeFragment.kt]: Done

File: app/src/main/java/com/example/foodrecipeapp/data/repo/source/remote/fetchjson/GetJsonFromUrl.kt (hunk 1 above)
Comment (Lomahs1401) [reply app/src/main/java/com/example/foodrecipeapp/data/repo/source/remote/fetchjson/GetJsonFromUrl.kt]: Done

**Review by ghost — APPROVED** [review]

**Pull request merged**